staging/src/k8s.io/apimachinery/pkg/runtime/serializer/cbor/internal/modes/decode_test.go

```diff
@@ -45,6 +45,257 @@ func TestDecode(t *testing.T) {
+			into: struct {
+				A int `json:"A"`
+			}{},
+			assertOnError: assertIdenticalError(&cbor.DupMapKeyError{Key: string("A"), Index: 1}),
+		},
+		{
+			name:          "reject duplicate text string keys into map",
+			modes:         []cbor.DecMode{modes.DecodeLax},
+			in:            hex("A2614101614102"), // {A:1, A:2}
+			into:          map[string]interface{}{},
+			assertOnError: assertIdenticalError(&cbor.DupMapKeyError{Key: string("A"), Index: 1}),
+		},
+		{
+			name:          "reject duplicate byte string keys into map",
+			in:            hex("A2414101414102"), // {'A':1, 'A':2}
+			into:          map[string]interface{}{},
+			assertOnError: assertIdenticalError(&cbor.DupMapKeyError{Key: string("A"), Index: 1}),
+		},
+		{
+			name:  "reject duplicate byte string keys into struct",
+			modes: []cbor.DecMode{modes.DecodeLax},
+			in:    hex("A2414101414102"), // {'A':1, 'A':2}
+			into: struct {
+				A int `json:"A"`
+			}{},
+			assertOnError: assertIdenticalError(&cbor.DupMapKeyError{Key: string("A"), Index: 1}),
+		},
+		{
+			name:          "reject duplicate byte string and text string keys into map",
+			in:            hex("A2414101614102"), // {'A':1, "A":2}
+			into:          map[string]interface{}{},
+			assertOnError: assertIdenticalError(&cbor.DupMapKeyError{Key: string("A"), Index: 1}),
+		},
+		{
+			name: "reject duplicate byte string & text string keys into struct",
+			in:   hex("A2414101614102"), // {'A':1, "A":2}
+			into: struct {
+				A int `json:"A"`
+			}{},
+			assertOnError: assertIdenticalError(&cbor.DupMapKeyError{Key: string("A"), Index: 1}),
+		},
+		{
+			name:  "reject duplicate byte string keys & one key with an unknown tag into struct",
+			modes: []cbor.DecMode{modes.DecodeLax},
+			in:    hex("A2D9EA63414101414102"), // {60003('A'):1, 'A':2}
+			into: struct {
+				A int `json:"A"`
+			}{},
+			assertOnError: assertIdenticalError(&cbor.DupMapKeyError{Key: string("A"), Index: 1}),
+		},
+		{
+			name:          "reject duplicate byte string keys & one key with unknown tag into map",
+			in:            hex("A2D9EA63414101414102"), // {60003('A'):1, 'A':2}
+			into:          map[string]interface{}{},
+			assertOnError: assertIdenticalError(&cbor.DupMapKeyError{Key: string("A"), Index: 1}),
+		},
+		{
+			name:  "map byte string key of type tag does not match struct field despite matching tag content",
+			modes: []cbor.DecMode{modes.DecodeLax},
+			in:    hex("A1D9EA63414101"), // {60003('A'):1}
+			into: struct {
+				A int `json:"A"`
+			}{},
+			assertOnError: assertOnConcreteError(func(t *testing.T, e *cbor.UnmarshalTypeError) {
+				expected_msg := "cbor: cannot unmarshal tag into Go value of type string (map key is of type tag and cannot be used to match struct field name)"
+				if e.Error() != expected_msg {
+					t.Errorf("expected %#v, got %#v", expected_msg, e.Error())
+				}
+			}),
+		},
+		{
+			name:  "map text string key of type tag does not match struct field despite matching tag content",
```

Comment:
```suggestion
			name:  "map key of type tag does not match struct field despite matching enclosed text string",
```

To be precise, in the CBOR data model the type of the key is "tag" and its "tag content" (the data item enclosed by the tag) has type "text string".